contributors/design-proposals/storage/local-storage-pv.md

```diff
@@ -761,3 +761,138 @@ The central manager will poll each worker for their discovered volumes and creat
 them.  When a PV is released, then it will send the cleanup request to the worker.
 
 Detailed design TBD
+
+
+### Phase 3: 1.11 Alpha
+
+#### Dynamic Provisioning
+
+The local volume provisioner will be enhanced to support dynamic provisioning of local storage,
+which depends on the newly introduced changes in [Volume Topology-aware Scheduling](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/volume-topology-scheduling.md),
+that is:
+
+1. When the feature gate `DynamicProvisioningScheduling` is enabled,
+the scheduler will verify a pod's volume requirements,
+and set an annotation `annScheduledNode` on PVCs that need provisioning.
+2. New fields `Capacity` will be added in `Status` of `StorageClass` to
+allow storage providers to report capacity limits per topology domain.
+
+The local volume provisioner needs to be updated in several areas to support dynamic provisioning.
+
+##### Backend storage management
+
+Mechanisms to manage local storage sources are varied. They could be LVM, Ceph, or something like that.
+In addition to current `populator`, `deletor` and `discoverer`, a new component named
+`storageManager` will be introduced in the plugin, to manage the backend storage.
+
+`storageManager` is a general interface, each kind of storage should have its own implementation.
+During start-up, a new environment variable `STORAGE_BACKEND` is set in the provisioner DaemonSet,
+to determine which StorageManager instance to start.
+(TBD: Should we support the scenario of multi backends on a node?)
```

Comment:
Wouldn't it be much simpler if a cluster operator just deploys multiple `DaemonSet`s, each for a different `STORAGE_BACKEND`?

Reply:
Yes, I didn't describe it clearly, I mean support multi backends in a same plugin, which seems no need :)